compiler/cgmeth.nim

```diff
@@ -229,17 +229,10 @@ proc genDispatcher(g: ModuleGraph; methods: seq[PSym], relevantCols: IntSet): PS
   var base = methods[0].ast[dispatcherPos].sym
   result = base
   var paramLen = sonsLen(base.typ)
-  var nilchecks = newNodeI(nkStmtList, base.info)
```

Comment:
Please do not remove nil checks for (multi-)methods, they are mandatory and must become part of the spec.

Reply:
because it can't determine the type? why is this different from a "normal" dereferencing where we segfault?

Reply:
Well yes. We also don't crash in ``if x of MyObject``, if ``x`` is nil the check is false.

Reply:
hm, but we don't raise an exception/`Defect` either - it's meaningful / useful to answer `false` without dereferencing.. 

in the case of dispatch, there seems to be no meaningful way to progress because the dereference is actually necessary to select an overload - thus panic/crash/raise seem equally appropriate.  this branch argues that it's better to behave the same way under all dereferences that can't be avoided (dispatch, accessing field etc) due to their nature.

Reply:
Well ... keep in mind that this was written for *multi* dispatch, it's perfectly fine if some arguments end up to be "nil" and yet other arguments can be used for dispatching.

Reply:
so should it raise then? ie if any argument is nil?

Reply:
basically, the existing code raises a `Defect` if any relevant argument in is `nil` - some arguments can't be nil.